pypesto/optimize/optimizer.py

```diff
@@ -167,6 +164,34 @@ def recover_result(objective, startpoint, err):
     return result
 
 
+def read_result_from_csv(problem: Problem, history_options: HistoryOptions,
```

Comment:
this is specific to the CsvHistory. Maybe as static method of CsvHistory?

Reply:
On second thoughts, it of course involves generation of an optimization result, which is not the job of CsvHistory. I guess it's fine here. Actually, it should be straightforward to write the same for Hdf5History. We can do that at some point, currently not yet needed for us. In that case, we would add `read_result_from_hdf5`.

Reply:
Should be straightforward to extend this method to account for other Histories based on suffix in storage file.